code/__byond_version_compat.dm

```diff
@@ -27,19 +27,39 @@
 #define LIBCALL call_ext
 #endif
 
-// So we want to have compile time guarantees these procs exist on local type, unfortunately 515 killed the .proc/procname syntax so we have to use nameof()
-#if DM_VERSION < 515
-/// Call by name proc reference, checks if the proc exists on this type or as a global proc
-#define PROC_REF(X) (.proc/##X)
+// So we want to have compile time guarantees these methods exist on local type, unfortunately 515 killed the .proc/procname and .verb/verbname syntax so we have to use nameof()
+
+// These are the generic defines that wrap for the either the sub-515 or post-515 syntax.
+
+/// Call by name proc references, checks if the proc exists on this type or as a global proc.
+#define PROC_REF(X) GENERIC_REF(##X, proc)
+/// Call by name verb references, checks if the verb exists on this type or as a global verb.
+#define VERB_REF(X) GENERIC_REF(##X, verb)
+
 /// Call by name proc reference, checks if the proc exists on given type or as a global proc
-#define TYPE_PROC_REF(TYPE, X) (##TYPE.proc/##X)
-/// Call by name proc reference, checks if the proc is existing global proc
-#define GLOBAL_PROC_REF(X) (/proc/##X)
+#define TYPE_PROC_REF(TYPE, X) TYPE_GENERIC_REF(##TYPE, ##X, proc)
+/// Call by name verb reference, checks if the verb exists on given type or as a global verb
+#define TYPE_VERB_REF(TYPE, X) TYPE_GENERIC_REF(##TYPE, ##X, verb)
+
+/// Call by name proc reference, checks if the proc is AN existing global proc
+#define GLOBAL_PROC_REF(X) GLOBAL_GENERIC_REF(##X, proc)
+/// Call by name verb reference, checks if the verb is AN existing global verb
+#define GLOBAL_VERB_REF(X) GLOBAL_GENERIC_REF(##X, verb)
```

Comment:
melbert mentioned this, these don't exist, we shouldn't have a macro for them it's misleading

Reply:
alrigh